classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
if the $key isset and the value is [false or null or empty],now:
Cache::getInstance()->exists($this->last_query_hash)//return true
Cache::getInstance()->get($this->last_query_hash)//return false

protected function _get($key)
{
/////....
$file = file_get_contents($filename);
return unserialize($file);
}

/**
 * @see Cache::_exists()
 */
protected function _exists($key)
{

/////....
return isset($this->keys[$key]) && @filemtime($this->getFilename($key));
}

so,Cache::getInstance()->exists($this->last_query_hash) is the real check whether there is cache.
Cache::getInstance()->get($this->last_query_hash) When the value is [false or null or empty], if the cache is used to determine whether there will cause bug